pkg/cli/cluster_free_disk_space.go

```diff
@@ -0,0 +1,133 @@
+
+	cmd := &cobra.Command{
+		Use:          "check-free-disk-space",
+		Short:        "Returns the amount of free disk space (in bytes) for a given storage class",
+		Example:      "kurl cluster check-free-disk-space --storageclass openebs",
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			classes, err := clientSet.StorageV1().StorageClasses().List(cmd.Context(), metav1.ListOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to list cluster storage classes: %w", err)
+			}
+
+			var selectedClass *v1.StorageClass
+			for _, class := range classes.Items {
+				val, ok := class.Annotations[isDefaultStorageClassAnnotation]
+				if ok && val == "true" && forStorageClass == "" {
+					selectedClass = &class
+					logger.Printf("Selected default storage class %q", selectedClass.Name)
+					break
+				}
+
+				if class.Name == forStorageClass {
+					selectedClass = &class
+					logger.Printf("Selected storage class %q", selectedClass.Name)
+					break
+				}
+			}
+
+			if selectedClass == nil {
+				if forStorageClass == "" {
+					return fmt.Errorf("failed to find default storage class")
+				}
+				return fmt.Errorf("failed to find storage class %q", forStorageClass)
+			}
+
+			switch selectedClass.Provisioner {
+			case openEBSLocalProvisioner:
+				freeSpaceGetter, err := clusterspace.NewOpenEBSFreeDiskSpaceGetter(clientSet, logger, openEBSImage, selectedClass.Name)
+				if err != nil {
+					fmt.Print(logbuf.String())
+					return fmt.Errorf("failed to start openebs free space getter: %w", err)
+				}
+
+				volumes, err := freeSpaceGetter.OpenEBSVolumes(cmd.Context())
+				if err != nil {
+					fmt.Print(logbuf.String())
+					return fmt.Errorf("failed to get openebs free space: %w", err)
+				}
+
+				for node, volume := range volumes {
+					if openEBSNode == "" {
+						fmt.Printf("%d\t%s\n", volume.Free, node)
+						continue
+					}
+
+					if node != openEBSNode {
+						logger.Printf("Skipping node %q", node)
+						continue
+					}
+
+					fmt.Printf("%d\n", volume.Free)
+					return nil
+				}
+
+				if openEBSNode != "" {
+					fmt.Print(logbuf.String())
+					return fmt.Errorf("failed to collect openebs free space: node %q not found", openEBSNode)
+				}
+				return nil
+
+			case rookCephFSProvisioner, rookRBDProvisioner:
+				freeSpaceGetter, err := clusterspace.NewRookFreeDiskSpaceGetter(clientSet, rookClientSet, selectedClass.Name)
+				if err != nil {
+					return fmt.Errorf("failed to start rook free space getter: %w", err)
+				}
+
+				space, err := freeSpaceGetter.GetFreeSpace(cmd.Context())
+				if err != nil {
+					return fmt.Errorf("failed to get rook free space: %w", err)
+				}
+
+				fmt.Println(space)
+				return nil
+
+			default:
+				return fmt.Errorf("provisioner %q is not supported", selectedClass.Provisioner)
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&forStorageClass, "storageclass", "", "The storage class for which to check the free disk space.")
```

Comment:
```suggestion
	cmd.Flags().StringVar(&forStorageClass, "storageclass", "", fmt.Sprintf("Inform the storage class for which to check the free disk space. [Options: %s, %s, %s]", openEBSLocalProvisioner, rookRBDProvisioner, rookCephFSProvisioner ))
```

	isDefaultStorageClassAnnotation = "storageclass.kubernetes.io/is-default-class"
	openEBSLocalProvisioner         = "openebs.io/local"
	rookRBDProvisioner              = "rook-ceph.rbd.csi.ceph.com"
	rookCephFSProvisioner           = "rook-ceph.cephfs.csi.ceph.com"